File: proyectoestacionamiento/lib/Core/api_config.dart
class ApiConfig {
  static const String baseUrl = String.fromEnvironment(
    'API_BASE_URL',
    defaultValue: 'https://smartparking-production-9a89.up.railway.app',
  );

  static const String companyPortalUrl = String.fromEnvironment(
    'COMPANY_PORTAL_URL',
    defaultValue: 'https://smart-parking-mlma.vercel.app/login',
  );

  static const String _parkingSocketOverride = String.fromEnvironment(
    'WS_PARKING_URL',
    defaultValue: '',
  );

  static String get parkingSocketUrl {
    if (_parkingSocketOverride.isNotEmpty) return _parkingSocketOverride;

    final apiUri = Uri.parse(baseUrl);
    final socketScheme = apiUri.scheme == 'https' ? 'wss' : 'ws';
    return apiUri.replace(scheme: socketScheme, path: '/ws/parking').toString();
  }

  static const String googleMapsApiKey = String.fromEnvironment(
    'GOOGLE_MAPS_API_KEY',
    defaultValue: 'AIzaSyDU9V_Db6UdM8vuvfQwSghwRdT3v-cOklk',
  );
}

File: estacionamientos-portal/src/app/core/api.config.ts
const LOCAL_BACKEND_PORT = '8000';
const RAILWAY_BACKEND_HOST = 'smartparking-production-9a89.up.railway.app';

const isPrivateNetworkHost = (hostname: string): boolean =>
  hostname === 'localhost' ||
  hostname === '127.0.0.1' ||
  hostname === '0.0.0.0' ||
  hostname.startsWith('192.168.') ||
  hostname.startsWith('10.') ||
  /^172\.(1[6-9]|2\d|3[01])\./.test(hostname);

const buildApiConfig = () => {
  const hostname = typeof window !== 'undefined' ? window.location.hostname : '';

  if (isPrivateNetworkHost(hostname)) {
    const localHost = hostname === '0.0.0.0' ? 'localhost' : hostname;

    return {
      baseUrl: `http://${localHost}:${LOCAL_BACKEND_PORT}`,
      parkingSocketUrl: `ws://${localHost}:${LOCAL_BACKEND_PORT}/ws/parking`,
    };
  }

  return {
    baseUrl: `https://${RAILWAY_BACKEND_HOST}`,
    parkingSocketUrl: `wss://${RAILWAY_BACKEND_HOST}/ws/parking`,
  };
};

export const API_CONFIG = buildApiConfig();
